R/start_transcription.R

```diff
@@ -27,14 +50,38 @@ function(
     format = tools::file_ext(url),
     language = "en-US",
     hertz = NULL,
+    output_bucket = NULL,
+    channel_identification = NULL,
+    show_speaker_labels = NULL,
+    max_speaker_labels = NULL,
+    vocabulary_name = NULL,
     ...
 ) {
     bod <- list(Media = list(MediaFileUri = url))
     bod$MediaFormat <- format
     bod$LanguageCode <- language
+    bod$TranscriptionJobName <- name
     if (!is.null(hertz)) {
         bod$MediaSampleRateHertz <- hertz
     }
-    bod$TranscriptionJobName <- name
+    if (!is.null(output_bucket)) {
+      bod$OutputBucketName <- output_bucket
+    }
+    # add on settings
+    if (any(
+      !is.null(channel_identification),
+      !is.null(show_speaker_labels),
+      !is.null(max_speaker_labels),
+      !is.null(vocabulary_name)
+      )) {
+      settings <- list()
```

Comment:
If you introduce a `more_settings` argument (with `list()` as default value), you can initialize with `settings <- more_settings` to incorporate extra settings and override them later if they're provided twice. In that case, it's worth adding a check `stopifnot(is.list(more_settings))` to prevent weird errors.